Lib/pathlib.py

```diff
@@ -562,15 +465,30 @@ def as_uri(self):
         """Return the path as a 'file' URI."""
         if not self.is_absolute():
             raise ValueError("relative path can't be expressed as a file URI")
-        return self._flavour.make_uri(self)
+
+        if self._flavour is posixpath:
```

Comment:
should not use `issubclass` ?

Reply:
`posixpath` is a module, not a class. It serves as the implementation for `os.path` for non-Windows OSs. See the second note in the [`os.path` docs](https://docs.python.org/3/library/os.path.html) for more info.

Unless you're suggesting that `issubclass` works with modules? From a tiny bit of testing that doesn't seem to be the case, but perhaps I'm missing something.

Reply:
FYI, pathlib began as a high-level *re-implementation* of `posixpath` and `ntpath`. This ceased to make sense when pathlib was merged into Python itself - why would we have two similar implementations of the same functionality in the same codebase? For the past few Python releases I've been merging these implementations, e.g. pathlib no longer re-implements `os.path.expanduser()` or `os.path.realpath()` -- instead, it wraps them. The result is that pathlib is becoming a high-level *wrapper* of `posixpath` and `ntpath`.

These `flavour is posixpath` expressions are the last vestiges of the pathlib-specific functionality that hasn't been merged back to `posixpath` and `ntpath`. If we want to remove them, we could introduce three new `os.path` functions and call through to them:

- `os.path.fileuri()` -- called by `pathlib.PurePath.as_uri()`
- `os.path.isreserved()` -- called by `pathlib.PurePath.is_reserved()`
- `os.path.splitroot()` -- called by `pathlib.PurePath._parse_args()`

Personally I'm on the fence about these.

Reply:
my bad @barneygale I had in mind that posixpath was a class here.